cmd/provider/main.go

```diff
@@ -62,6 +71,17 @@ func main() {
 
 	rl := ratelimiter.NewDefaultProviderRateLimiter(ratelimiter.DefaultProviderRPS)
 	kingpin.FatalIfError(apis.AddToScheme(mgr.GetScheme()), "Cannot add AWS APIs to scheme")
-	kingpin.FatalIfError(controller.Setup(mgr, log, rl), "Cannot setup AWS controllers")
+	kingpin.FatalIfError(controller.Setup(mgr, log, rl, getProviderSetupFn(*providerSource, *providerVersion), *concurrency), "Cannot setup AWS controllers")
 	kingpin.FatalIfError(mgr.Start(ctrl.SetupSignalHandler()), "Cannot start controller manager")
 }
+
+func getProviderSetupFn(source, version string) terraform.ProviderSetupFn {
```

Comment:
Can we make `clients.ProviderConfigurationBuilder` to contain the content of this function to simplify the flow? Like
```go
package clients

func ProviderConfigurationBuilder(source, version string) ProviderConfigFn {
  // Note that it doesn't return []byte anymore.
  return func(ctx context.Context, client client.Client, mg xpresource.Managed) (ProviderConfiguration, error) {
		pr := tfcli.ProviderRequirement{
			Source:  source,
			Version: version,
		}
		// original content of ProviderConfigBuilder function.
		pr.Credentials = <calculated credentials>
		return pr, err
	}
}
```